src/rewards.js

```diff
@@ -17,15 +17,58 @@ rewards.YOUTUBE_CREATOR = 'youtube_creator';
 rewards.TYPE_DAILY_VIEW = 'daily_view';
 
 rewards.claimReward = (type, rewardParams) => {
+  let reason;
+  switch (type) {
```

Comment:
making these a function is a little cleaner, since it makes the break statements unnecessary